subprojects/docs/src/docs/userguide/publishingMaven.adoc

```diff
@@ -300,3 +300,48 @@ The result is that the following artifacts will be published:
 * The Javadoc JAR artifact that has been explicitly configured: `my-library-1.0-javadoc.jar`
 
 The <<signing_plugin, Signing Plugin>> is used to generate a signature file for each artifact. In addition, checksum files will be generated for all artifacts and signature files.
+
+[[publishing_maven:deferred_configuration]]
+=== Removal of deferred configuration behavior
+
+Until Gradle 4.7, the `publishing {}` block was implicitly treated as if all the logic inside it was executed after the project is evaluated.
+This caused quite a bit of confusion, because it was the only block that behaved that way.
+As part of the stabilization effort in Gradle 4.8, we are deprecating this behavior and asking all users to migrate their build.
+
+The new, stable behavior can be switched on by adding the following to your settings file:
+
+    enableFeaturePreview('STABLE_PUBLISHING')
+
+We recommend doing a test run with a local repository to see whether all artifacts still have the expected coordinates.
+In most cases everything should work as before and you are done.
+
+If the coordinates change unexpectedly, you may have some logic inside your publishing block or in a plugin that is depending on the deferred configuration behavior.
+For instance, the following logic assumes that the subprojects will be evaluated when the artifactId is set:
+
+    subprojects {
+        publishing {
+            publications {
+                mavenJava {
+                    from components.java
+                    artifactId = jar.baseName
+                }
+            }
+        }
+    }
+
+This kind of logic must be wrapped in an `afterEvaluate {}` block to make it work going forward.
+
+    subprojects {
+        publishing {
+            publications {
+                mavenJava {
+                    from components.java
+                    afterEvaluate {
+                        artifactId = jar.baseName
+                    }
+                }
+            }
+        }
+    }
```

Comment:
The code blocks should use 

```
[source,groovy]
----
...
----
```

to get syntax highlighting.

Reply:
Done